### PR TITLE
Inherit Brave Ads .clang-tidy checks from the parent config

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/27935

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by aseren — APPROVED** [review]

File: components/brave_ads/browser/component_updater/component_util.cc [hunk 1]
```diff
@@ -10,8 +10,8 @@
 namespace brave_ads {
 
 absl::optional<ComponentInfo> GetComponentInfo(const std::string& id) {
-  const auto iter = components.find(id);
-  if (iter == components.end()) {
+  const auto iter = kComponents.find(id);
```
Comment (goodov) [inline components/brave_ads/browser/component_updater/component_util.cc:13]: holy moly. the huge `kComponents` vector declared in `components/brave_ads/browser/component_updater/component_util.h` is crazy! please move it into `.cc` file as constexpr or as `base::NoDestructor` accessible via `GetComponent()` global function.

File: components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc [hunk 2]
```diff
@@ -75,7 +75,7 @@ class TestWebPageConstructor final {
   }
 
   schema_org::mojom::EntityPtr CreateCreativeEntity() {
-    const char* kSearchResultAdStringAttributes[] = {
+    const char* search_result_ad_string_attributes[] = {
```
Comment (goodov) [inline components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc:78]: `constexpr const char*`

File: vendor/bat-native-ads/src/bat/ads/internal/account/statement/ad_rewards_features_unittest.cc [hunk 3]
```diff
@@ -45,10 +45,10 @@ TEST(BatAdsAdRewardsFeaturesTest, AdRewardsDisabled) {
 TEST(BatAdsAdRewardsFeaturesTest, AdRewardsNextPaymentDay) {
   // Arrange
   std::vector<base::test::FeatureRefAndParams> enabled_features;
-  base::FieldTrialParams kAdRewardsParameters;
-  const char kNextPaymentDayParameter[] = "next_payment_day";
-  kAdRewardsParameters[kNextPaymentDayParameter] = "5";
-  enabled_features.emplace_back(kAdRewards, kAdRewardsParameters);
+  base::FieldTrialParams ad_rewards_parameters;
+  const char next_payment_day_parameter[] = "next_payment_day";
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/account/statement/ad_rewards_features_unittest.cc:49]: `constexpr char`

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/permission_rules/user_activity_permission_rule_unittest.cc [hunk 4]
```diff
@@ -21,12 +21,12 @@ class BatAdsUserActivityPermissionRuleTest : public UnitTestBase {
     UnitTestBase::SetUp();
 
     base::FieldTrialParams parameters;
-    const char kTriggersParameter[] = "triggers";
-    parameters[kTriggersParameter] = "0D=1.0;0E=1.0;08=1.0";
-    const char kTimeWindowParameter[] = "time_window";
-    parameters[kTimeWindowParameter] = "1h";
-    const char kThresholdParameter[] = "threshold";
-    parameters[kThresholdParameter] = "2.0";
+    const char triggers_parameter[] = "triggers";
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/permission_rules/user_activity_permission_rule_unittest.cc:24]: `constexpr char` everything here.

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/targeting/top_segments_unittest.cc [hunk 5]
```diff
@@ -163,12 +163,12 @@ TEST_P(BatAdsTopSegmentsTest, GetSegments) {
   std::vector<base::test::FeatureRef> disabled_features;
 
   if (param.epsilon_greedy_bandits_enabled) {
-    const char kEpsilonValue[] = "epsilon_value";
-    base::FieldTrialParams kEpsilonGreedyBanditParameters;
+    const char epsilon_value[] = "epsilon_value";
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/targeting/top_segments_unittest.cc:166]: `constexpr char`

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/targeting/top_segments_unittest.cc [hunk 6]
```diff
@@ -236,15 +236,15 @@ TEST_F(BatAdsTopSegmentsTest, GetSegmentsForAllModelsIfPreviouslyProcessed) {
   ProcessTextClassification();
   ProcessPurchaseIntent();
 
-  const char kEpsilonValue[] = "epsilon_value";
-  std::map<std::string, std::string> kEpsilonGreedyBanditParameters;
+  const char epsilon_value[] = "epsilon_value";
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/targeting/top_segments_unittest.cc:239]: `constexpr char`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/debug/debug_command_line_switch_parser_util_unittest.cc [hunk 7]
```diff
@@ -22,15 +22,16 @@ constexpr char kRewardsSwitch[] = "rewards";
 struct ParamInfo final {
   CommandLineSwitchInfo command_line_switch;
   bool expected_should_debug;
-} kTests[] = {{/*command_line_switch*/ {kRewardsSwitch, "debug=true"},
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/debug/debug_command_line_switch_parser_util_unittest.cc:25]: should be `const`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_command_line_switch_values_util_unittest.cc [hunk 8]
```diff
@@ -27,7 +27,7 @@ constexpr char kFooBarSwitch[] = "foobar";
 struct ParamInfo final {
   CommandLineSwitchInfo command_line_switch;
   bool expected_did_override_command_line_switch;
-} kTests[] = {
+} g_k_tests[] = {
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_command_line_switch_values_util_unittest.cc:30]: `const`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_command_line_switches_util_unittest.cc [hunk 9]
```diff
@@ -27,10 +27,10 @@ constexpr char kEnableAutomationSwitch[] = "enable-automation";
 struct ParamInfo final {
   CommandLineSwitchInfo command_line_switch;
   bool expected_did_override_command_line_switch;
-} kTests[] = {{/*command_line_switch*/ {kFooBarSwitch, {}},
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_command_line_switches_util_unittest.cc:30]: `const`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_features_from_command_line_util_unittest.cc [hunk 10]
```diff
@@ -42,7 +42,7 @@ constexpr char kFooBarSwitch[] = "foobar";
 struct ParamInfo final {
   CommandLineSwitchInfo command_line_switch;
   bool expected_did_override_from_command_line;
-} kTests[] = {
+} g_k_tests[] = {
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/did_override/did_override_features_from_command_line_util_unittest.cc:45]: `const`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/environment/environment_command_line_switch_parser_util_unittest.cc [hunk 11]
```diff
@@ -25,16 +25,16 @@ constexpr char kRewardsSwitch[] = "rewards";
...
-               /*expected_environment_type*/ EnvironmentType::kStaging},
-              {/*command_line_switch*/ {kRewardsSwitch, "staging=1"},
-               /*expected_environment_type*/ EnvironmentType::kStaging},
-              {/*command_line_switch*/ {kRewardsSwitch, "staging=false"},
-               /*expected_environment_type*/ EnvironmentType::kProduction},
-              {/*command_line_switch*/ {kRewardsSwitch, "staging=foobar"},
-               /*expected_environment_type*/ EnvironmentType::kProduction},
-              {/*command_line_switch*/ {},
-               /*expected_environment_type*/ kDefaultEnvironmentType}};
+} g_k_tests[] = {{/*command_line_switch*/ {kRewardsSwitch, "staging=true"},
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/environment/environment_command_line_switch_parser_util_unittest.cc:28]: `const`

File: vendor/bat-native-ads/src/bat/ads/internal/flags/flag_manager_unittest.cc [hunk 12]
```diff
@@ -33,7 +33,7 @@ struct ParamInfo final {
   bool expected_should_debug;
   bool expected_did_override_command_line_switches;
   EnvironmentType expected_environment_type;
-} kTests[] = {
+} g_k_tests[] = {
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/flags/flag_manager_unittest.cc:36]: `const`

File: vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc [hunk 13]
```diff
@@ -29,19 +29,19 @@ TEST_F(BatAdsVectorDataTest, DenseVectorDataInitialization) {
 
 TEST_F(BatAdsVectorDataTest, SparseVectorDataInitialization) {
   // Arrange
-  const int kDimensionCount = 6;
+  const int dimension_count = 6;
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc:32]: `constexpr int`

File: vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc [hunk 14]
```diff
@@ -29,19 +29,19 @@ TEST_F(BatAdsVectorDataTest, DenseVectorDataInitialization) {
...
 
   // Assert
-  EXPECT_EQ(kDimensionCount, sparse_data_vector_6.GetDimensionCount());
+  EXPECT_EQ(dimension_count, sparse_data_vector_6.GetDimensionCount());
 }
 
 TEST_F(BatAdsVectorDataTest, DenseDenseProduct) {
   // Arrange
-  const double kTolerance = 1e-6;
+  const double tolerance = 1e-6;
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc:44]: `constexpr double`

File: vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc [hunk 15]
```diff
@@ -58,14 +58,14 @@ TEST_F(BatAdsVectorDataTest, DenseDenseProduct) {
...
-              std::fabs(6.0 - res_3x1) < kTolerance);
+  EXPECT_TRUE(std::fabs(14.0 - res_3x3) < tolerance &&
+              std::fabs(55.0 - res_5x5) < tolerance &&
+              std::fabs(6.0 - res_3x1) < tolerance);
 }
 
 TEST_F(BatAdsVectorDataTest, SparseSparseProduct) {
   // Arrange
-  const double kTolerance = 1e-6;
+  const double tolerance = 1e-6;
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc:68]: same here and below

File: vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc [hunk 16]
```diff
@@ -69,7 +69,7 @@ TEST_F(BatAdsLinearTest, BiasesPredictionTest) {
 
 TEST_F(BatAdsLinearTest, BinaryClassifierPredictionTest) {
   // Arrange
-  const size_t kExpectedPredictionSize = 1;
+  const size_t expected_prediction_size = 1;
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc:72]: `constexpr size_t`

File: vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc [hunk 17]
```diff
@@ -87,16 +87,16 @@ TEST_F(BatAdsLinearTest, BinaryClassifierPredictionTest) {
...
+  ASSERT_EQ(expected_prediction_size, predictions_1.size());
 
   EXPECT_TRUE(predictions_0.at("the_only_class") < 0.5 &&
               predictions_1.at("the_only_class") > 0.5);
 }
 
 TEST_F(BatAdsLinearTest, TopPredictionsTest) {
   // Arrange
-  const size_t kPredictionLimits[2] = {2, 1};
+  const size_t prediction_limits[2] = {2, 1};
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc:99]: `constexpr`

File: vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/text_processing_unittest.cc [hunk 18]
```diff
@@ -44,9 +44,9 @@ class BatAdsTextProcessingTest : public UnitTestBase {};
 
 TEST_F(BatAdsTextProcessingTest, BuildSimplePipeline) {
   // Arrange
-  const double kTolerance = 1e-6;
-  const unsigned kExpectedLen = 3;
-  const std::string kTestString = "Test String";
+  const double tolerancee = 1e-6;
+  const unsigned expected_len = 3;
+  const std::string test_string = "Test String";
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/text_processing_unittest.cc:49]: `constexpr` numbers. std::string can be converted to `constexpr char`.

**Review by goodov — COMMENTED** [review]
I assume you did an automatic replacement, but it's better to convert POD-type constants to `constexpr` and in general just use `contsexpr` everywhere so we can use `kStyle` variable naming for true compile-time constants, and fallback to `lower_case` for runtime constants that should be dynamically constructed such as `std::vector`.

`static const` and `constexpr` are styled to be `kConstant`, but `const <type> a = ...;` variables are styled to be `lower_case`. Here's some explanation in the description, pls see `LocalConstantCase` block: https://github.com/brave/brave-core/pull/16726

**Pull request merged**